hugr-core/src/types.rs

```diff
@@ -148,10 +148,16 @@ impl std::fmt::Display for SumType {
         }
 
         match self {
+            SumType::Unit { size: 1 } => write!(f, "Unit"),
+            SumType::Unit { size: 2 } => write!(f, "Bool"),
             SumType::Unit { size } => {
                 display_list_with_separator(repeat_n("[]", *size as usize), f, "+")
```

Comment:
I think this should use something other than square brackets since they are also used for type parameters/extensions

Reply:
Type rows always get printed as `[t1, t2, ...]` since they are a _list_ of types. Should we change that?

Reply:
that makes sense, but a sum holding a empty rows shouldn't just be `[]+[]+...` right? Maybe just put in parens? Or something like `([]|[]|..)` ?

Reply:
Currently the sum of three rows where a single one is empty gets printed as
`[T]+[]+[T]`
if we empty them we get `Sum::Unit{3}`, displayed as
`[]+[]+[]`
adding parenthesis anywhere would be inconsistent with the non-empty case and it would look noisy when printed (keep in mind this is a `Display` impl, so being easy to read is more important that not being ambiguous).